### DISCUSSION TITLE
API: how to query node whether it's a directory (in a consistent way)?

### DISCUSSION THREAD

**Discussion by przepompownia** [nvim-tree/nvim-tree.lua#2443] [Q&A] [answered]
https://github.com/vim/vim/issues/12087#issuecomment-1517808746 is still not fixed so I need to use some custom replacement for `:tab drop`. 

Currently in `on_attach` I have to recognize directories by looking into details
```lua
    vim.keymap.set(
      'n',
      '<CR>',
      injectNode(function(node)
        if node.name == '..' or node.type == 'directory' then
          treeapi.node.open.edit()

          return
        end
        treeapi.tree.close()
        base.tab_drop_path(node.absolute_path)
      end),
      {buffer = bufnr, noremap = true}
    )
```

It would be good to have `is_directory()` directly on each node.

**Comment by Akmadan23** [comment, marked as answer]
I'm not sure what you're looking for, nodes are just tables, so to check whether a node is a directory you have two options: 
- check if `node.type` is `'directory'` (like you did)
- check if `node.nodes` is a table

Then there is the root node, which actually is not a node, since it's a table with only the `name` field.

**Reply by przepompownia** [reply]
I'm looking for more clean method but (as you suggest too) it seems that currently we have to guess by internals of `node`.

**Reply by Akmadan23** [reply]
Yes, if you really want a cleaner way to write it you can create your own local function that does it so that you don't have to write the same things several times.
